examples/cst/cst_NodeSelector.py

```diff
@@ -18,23 +24,25 @@ class NodeSelector(cst.CSTVisitor):
     """
 
     def __init__(self) -> None:
-        self.pandas_imported: bool = False
-        self.pandas_star_imported: bool = True  # TODO: use this in code
-        self.pandas_aliases = ["pandas"]
-        self.imported_pandas_aliases: list[cst.ImportAlias] = []
 
         self.variables: dict[str, Node] = {}
+        """Maps the library names in namespace to the responsible InputModule"""
+        self.libraries: dict[str, InputModule] = {}
+        """Maps the method names in namespace to the responsible InputModule and the original method name."""
+        self.library_methods: dict[str, Tuple[InputModule, str]] = {}
 
         super().__init__()
 
     def generic_visit(self, cst_node: CSTNode) -> Node:
         class_name = str(cst_node.__class__.__name__).replace("CST", "")
-        method_name = f"visit_{class_name}"
+        method_name = f"parse_{class_name}"
```

Comment:
es gab vorher kein Grund diese Methoden als visit Methoden zu bezeichnen. Das lässt sie nämlich aussehen wie overrides der Basisklasse, was sie überhaupt nicht sind. Im Grunde haben wir hier nämlich ein eigenes kleines visitor pattern innerhalb unseres visitors implementiert. Vielleicht greift man das in Zukunft auf und nutzt richtige CSTVisitors innerhalb unseres visitors. Vielleicht würde das den code schöner machen

Reply:
Naja, ueberschrieben haben sie die visit methoden ja schon, nur mit side-effects, die uns vorher nicht so bewusst waren. 

Keine Ahnung, ob ein visitor im visitor so hilfreich ist. Wir muessen das Teil erstmal auf die Strasse bekommen, bevor wir mit solchen Optimierungen anfangen...